### PR TITLE
ethdb/pebble: handle error in pebble module

### PR DESCRIPTION
**Pull request by mask-pp**
- Handle error in pebble module
- Use `common.CopyBytes`

### CODE REVIEW

File: ethdb/pebble/pebble.go
Comment (holiman) [inline ethdb/pebble/pebble.go:316]: From the docs on `db.Get` in pebble/db.go:
```
// ... The returned
// slice will remain valid until the returned Closer is closed. On success, the
// caller MUST call closer.Close() or a memory leak will occur.
```
You changed the order here, and call `Close` before we're finished operating on the returned slice. This PR thus subtly breaks the database api. 

I don't really see any point in this PR. If there's an error in the db, which surfaces on Close here, then it's probably pretty bad (someone yanked the disk?) and it'll wreak havoc somewhere else anyway

Comment (holiman) [reply ethdb/pebble/pebble.go:316]: OTOH, sure, I can agree that it's probably "The Right Thing To Do", to handle errors where they appear. But please make no other changes. Using `common.CopyBytes` here is not great, IMO. Keep the `copy` as is, rather than hope for `common.CopyBytes` to become inlined magically.

File: ethdb/pebble/pebble.go
Comment (holiman) [inline ethdb/pebble/pebble.go:526]:
```suggestion
	return err
```
Yes I know it can be done like you did it, but we generally don't do it that way. 
It's ok to declare the outputs if you want to make it obvious what each return means, for example if the return is `bool, bool, bool`, it's nicer if it says `exists, foo, existedEarlier bool`. 

However, that doesn't apply here. So just do 
```
err := b.b....
b.size ...
return err
```

File: ethdb/pebble/pebble.go
Comment (mask-pp) [reply ethdb/pebble/pebble.go:316]: Great catch, copy operation should done before `Close`. Will be reverted.

File: ethdb/pebble/pebble.go
Comment (mask-pp) [reply ethdb/pebble/pebble.go:526]: Will change it.

**Review by karalabe — APPROVED** [review]
SGTM, can emrge after release

**Review by holiman — APPROVED** [review]
LGTM, but let's wait until after a release before merging it

**Pull request merged**
